addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/HueEmulationUpnpServer.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.io.hueemulation.internal;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.MulticastSocket;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Advertises a Hue UPNP compatible bridge
+ *
+ * @author Dan Cunningham
+ *
+ */
+public class HueEmulationUpnpServer extends Thread {
+    private Logger logger = LoggerFactory.getLogger(HueEmulationUpnpServer.class);
+
+    static final private int UPNP_PORT = 1900;
+    static final private String MULTI_ADDR = "239.255.255.250";
+    private MulticastSocket socket;
+    private boolean running;
+    private String discoPath;
+    private String usn;
+
+    private String discoString = "HTTP/1.1 200 OK\r\n" + "CACHE-CONTROL: max-age=86400\r\n" + "EXT:\r\n"
+            + "LOCATION: %s\r\n" + "OPT: \"http://schemas.upnp.org/upnp/1/0/\"; ns=01\r\n"
+            + "01-NLS: 00000000-722C-4995-89F2-ABCDEF000000\r\n" + "ST: urn:schemas-upnp-org:device:basic:1\r\n"
+            + "USN: uuid:%s\r\n\r\n";
+
+    /**
+     * Server to send UDP packets onto the network when requested by a Hue API compatible device.
+     *
+     * @param discoPath
+     *            The URI path where the discovery xml document can be retrieved
+     * @param usn
+     *            The unique USN id for this server
+     */
+    public HueEmulationUpnpServer(String discoPath, String usn) {
+        this.running = true;
+        this.discoPath = discoPath;
+        this.usn = usn;
+    }
+
+    /**
+     * Stops the upnp server from running
+     */
+    public void shutdown() {
+        this.running = false;
+    }
+
+    @Override
+    public void run() {
+        while (running) {
+            try {
+                InetSocketAddress socketAddr = new InetSocketAddress(MULTI_ADDR, UPNP_PORT);
+                socket = new MulticastSocket(UPNP_PORT);
```

Comment:
Shouldn't you actually close this socket again, when leaving the `while(running)` loop?